templates/core/terraform/api-webapp/api-webapp.tf

```diff
@@ -0,0 +1,215 @@
+resource "azurerm_app_service_plan" "core" {
+  name                = "plan-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  reserved            = true
+  kind                = "linux"
+  sku {
+    tier     = "PremiumV3"
+    capacity = 1
+    size     = "P1v3"
+  }
+}
+
+resource "azurerm_application_insights" "core" {
+  name                = "appi-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  application_type    = "web"
+}
+
+resource "azurerm_app_service" "management_api" {
+  name                = "api-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  app_service_plan_id = azurerm_app_service_plan.core.id
+
+  identity {
+    type         = "UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.management_api.id]
+  }
+
+  https_only = true
+  app_settings = {
+
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = azurerm_application_insights.core.instrumentation_key
+  }
+
+  site_config {
+    app_command_line            = "gunicorn -w 2 -k uvicorn.workers.UvicornWorker main:app"
```

Comment:
What's this doing at this point? Not required if using a docker image. Guess need to decide on an approach.

Reply:
Since terraform 0.14 dependency lock file is recommended to be committed to git, as per:
https://www.terraform.io/docs/language/dependency-lock.html

Reply:
It's essentially cutting a step from github action on CI/CD, once we hook it up. This can be replaced once the container approach is built.